Make formula more iam friendly

filipposc5 opened the pull request:
> ### Create IAM friendly custom dup-iam handler

 Creates an IAM friendly custom `dup-iam` handler based on the existing `dup`:

From /usr/share/backupninja:

```
0a1
>
49,51c50,53
< if [ "`echo $desturl | /usr/bin/awk -F ':' '{print $1}'`" == "s3+http" ]; then
<    [ -n "$awsaccesskeyid" -a -n "$awssecretaccesskey" ]  || fatal "AWS access keys must be set for S3 backups."
< fi

---
> # Ditch this
> # if [ "`echo $desturl | /usr/bin/awk -F ':' '{print $1}'`" == "s3+http" ]; then
> #    [ -n "$awsaccesskeyid" -a -n "$awssecretaccesskey" ]  || fatal "AWS access keys must be set for S3 backups."
> # fi
251,253c253,255
< if [ "`echo $desturl | /usr/bin/awk -F ':' '{print $1}'`" == "s3+http" ]; then
<    export AWS_ACCESS_KEY_ID="$awsaccesskeyid"
<    export AWS_SECRET_ACCESS_KEY="$awssecretaccesskey"

---
> if [ ! -z "${awsaccesskeyid}" ] && [ "`echo $desturl | /usr/bin/awk -F ':' '{print $1}'`" == "s3+http" ]; then
>    export AWS_ACCESS_KEY_ID="${awsaccesskeyid:-}"
>    export AWS_SECRET_ACCESS_KEY="${awssecretaccesskey:-}"
357a360
>
```

These files are identical: dup-iam.helper dup.helper. The dup handler was
referenced from backupninja 1.0.1-2

> ### Make duplicity daily helper more IAM friendly

When using IAM policies the AWS env vars are no longer needed.

filipposc5 committed:
Make duplicity daily helper more IAM friendly

When using IAM policies the AWS env vars are no longer needed.

filipposc5 committed:
Updates dup-iam handler to be more IAM friendly

Updates dup-iam to be more IAM friendly by not erroring out if no AWS
keys have been set, and no longer attempts to set the env vars, if they
are empty so that duplicity stops giving this error:

`BackendException: No connection to backend`

Also in this commit:
* renames:    backupninja/files/dup.helper -> backupninja/files/dup-iam.helper

filipposc5 committed:
Update Changelog and readme

SteveMarshall commented:
Per our discussion on Hipchat, this would probably be clearer if there was an explicit commit of the unedited `dup`, followed by a commit with the modifications. Once you've got that, I'll give it a proper review 👍

filipposc5 committed:
Add original dup and dup.helper

Add the original duplicity handler and helper which will be the starting point
of creating a new duplicity handler. The files are taken directly from:
Ubuntu 14.04.4 LTS (trusty), backupninja 1.0.1-2, /usr/share/backupninja/

filipposc5 committed:
Rename dup handler to dup-iam

This effectively creates a new handler called dup-iam, whilst leaving the old
`dup` handler unchanged. Doing this in separate steps to increase visibility.

filipposc5 commented:
Restructured the commits and comments so they can tell the story with more clarity.
